src/libcore/hash/sip.rs

```diff
@@ -19,15 +19,17 @@ use super::Hasher;
 ///
 /// See: http://131002.net/siphash/
 ///
-/// Consider this as a main "general-purpose" hash for all hashtables: it
-/// runs at good speed (competitive with spooky and city) and permits
-/// strong _keyed_ hashing. Key your hashtables from a strong RNG,
-/// such as `rand::Rng`.
+/// This is the default hashing function used by standard library (eg.
+/// `collections::HashMap` uses it by default).
 ///
-/// Although the SipHash algorithm is considered to be cryptographically
-/// strong, this implementation has not been reviewed for such purposes.
-/// As such, all cryptographic uses of this implementation are strongly
-/// discouraged.
+/// SipHash is a general-purpose hashing function: it runs at a good
+/// speed (competitive with Spooky and City) and permits strong _keyed_
+/// hashing. This lets you key your hashtables from a strong RNG, such
+/// as [`rand::Rng`](https://doc.rust-lang.org/rand/rand/trait.Rng.html).
```

Comment:
Unfortunately all implementors of `Rng` aren't necessarily strong RNGs. For example xorshift is an implementation of this trait, and I believe it's not necessarily considered a strong RNG. Maybe the "such as" phrase should be dropped here?